### PR TITLE
VZ-9020: Create experimental CAPI install component

### PR DESCRIPTION
**Pull request by bobdonat**
This pull request implements the first iteration of the CAPI install component. Installing CAPI required a valid kubeconfig which needed to be generated when starting the VPO.

To install the CAPI component:
1. Install Verrazzanno.  The Cert Manager should be installed.
2. Create a secret called oci in the verrazzano-install namespace.  The secret contains OCI credentials which are currently required for the CAPI OCI provider.  The secret can be created with the script at platform-operator/scripts/install/create_oci_config_secret.sh.
3. Create a configMap to initiate the install of the CAPI component
```
kubectl apply -f - <<EOF
apiVersion: v1
data:
  name: verrazzano-capi
kind: ConfigMap
metadata:
  labels:
    experimental.verrazzano.io/configmap-kind: ShimComponent
    experimental.verrazzano.io/configmap-apiversion: v1beta2
  name: dev-capi
  namespace: default
EOF
``` 
To uninstall the CAPI component
```
kubectl delete cm dev-capi
```

### CODE REVIEW

File: platform-operator/thirdparty/charts/rancher/templates/service.yaml [hunk 1]
```diff
@@ -16,3 +16,18 @@ spec:
     name: https-internal
   selector:
     app: {{ template "rancher.fullname" . }}
+---
```
Comment (mgianatagh) [inline platform-operator/thirdparty/charts/rancher/templates/service.yaml:19]: This is already in branch anders/rancher-master.  Has the branch for this PR been rebased recently?

File: tests/e2e/verify-infra/restapi/rancher_url_test.go [hunk 2]
```diff
@@ -94,18 +94,6 @@ var _ = t.Describe("rancher", Label("f:infra-lcm",
 				metrics.Emit(t.Metrics.With("get_local_authconfig_state_elapsed_time", time.Since(start).Milliseconds()))
 
 				if minVer14 {
-					start = time.Now()
```
Comment (mgianatagh) [inline tests/e2e/verify-infra/restapi/rancher_url_test.go:97]: this code was removed from this branch a while ago.

File: platform-operator/capi/.cluster-api/clusterctl.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+images:
+  control-plane-ocne:
+    # TODO: these should come from the BOM
+    repository: ghcr.io/verrazzano
```
[the remark this answers is not in the document]
Comment (bobdonat) [reply platform-operator/capi/.cluster-api/clusterctl.yaml:7]: yes, we cannot use the BOM with an experimental component

File: platform-operator/thirdparty/charts/rancher/templates/service.yaml (hunk 1 above)
Comment (bobdonat) [reply platform-operator/thirdparty/charts/rancher/templates/service.yaml:19]: not sure what is wrong here

Comment (bobdonat) [reply platform-operator/thirdparty/charts/rancher/templates/service.yaml]: fixed

File: tests/e2e/verify-infra/restapi/rancher_url_test.go (hunk 2 above)
Comment (bobdonat) [reply tests/e2e/verify-infra/restapi/rancher_url_test.go]: fixed

**Review by mgianatagh — APPROVED** [review]

**Review by mgianatagh — APPROVED** [review]

**Pull request merged**